codec/msgpack.go

```diff
@@ -286,6 +286,15 @@ func (e *msgpackEncDriver) EncodeFloat64(f float64) {
 }
 
 func (e *msgpackEncDriver) EncodeTime(t time.Time) {
+	// use the MarshalBinary format if requested
+	if e.h.TimeNotBuiltin {
```

Comment:
IIUC correctly we don't need to concern ourselves with this setting in the decode function because it handles both cases seamlessly right?

Reply:
Correct.